### ISSUE TITLE
Compilation error on vdl-linkmap.c on Debian Jessie

### ISSUE THREAD

**Issue by frochet** [shadow/shadow#404]
Hi Rob,

I gave a try to the current master branch of Shadow and Shadow-tor-plugin on my Debian Jessie. I first had a build error on the elf-loader part of shadow (using gcc (Debian 4.9.2-10+deb8u1) 4.9.2), which was solved by adding the flag -std=gnu99. I suppose the bug is related to your commit 62ea55986733676fb3c48fa919c5a8f02e2960a1 on Shadow repo. Here is the line:

shadow/src/external/elf-loader/vdl-linkmap.c:208:3: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode

**Comment by robgjansen** [comment]
@jtracey ahh, did I mess this up? The commit referenced above was done in order to compile this on cent os 7, but I guess that broke the debian build? Sorry!

**Comment by jtracey** [comment]
@robgjansen If I'm reading the error correctly, no, I think you fixed this.

@frochet Does the version of Shadow you're compiling actually contain that commit?

**Comment by frochet** [comment]
@jtracey: Yes, the commit above is on the master branch. And I compiled today on the master. I was probably unclear: I've added -std=gnu99 flag on my side to compile successfully, but I guess there is something better to do?

**Comment by jtracey** [comment]
Well, it's just that the error you posted refers to a for loop on line 208 with a declaration, when the for loop is on line 209 after that commit, and it moves the declaration out, making it valid C89 again.

**Comment by frochet** [comment]
Oh haha. I was on the wrong origin, not having this commit indeed. I was using 3-days old origin, missing the fix. I got confused, sorry :/
